daemon/cluster/listen_addr.go

```diff
@@ -87,6 +89,39 @@ func (c *Cluster) resolveAdvertiseAddr(advertiseAddr, listenAddrPort string) (st
 	return systemAddr.String(), listenAddrPort, nil
 }
 
+// validateDefaultAddrPool validates default address pool
+// it also strips white space from the string before validation
+func validateDefaultAddrPool(defaultAddrPool []string, size uint32) error {
+	if defaultAddrPool == nil {
+		// defaultAddrPool is not defined
+		return nil
+	}
+	//if size is not set, then we use default value 24
+	if size == 0 {
+		size = 24
+	}
+	if size > 32 {
+		return fmt.Errorf("subnet size is out of range: %d", size)
+	}
+	for i := range defaultAddrPool {
+		//trim leading and trailing white spaces
+		defaultAddrPool[i] = strings.TrimSpace(defaultAddrPool[i])
+		result := strings.Split(defaultAddrPool[i], "/")
+		if len(result) > 2 {
+			return fmt.Errorf("Error in CIDR format : %s", defaultAddrPool[i])
+		}
+		// get the size from the string
+		ones, err := strconv.Atoi(result[1])
+		if err != nil {
+			return fmt.Errorf("Error in getting size from CIDR format : %s", defaultAddrPool[i])
+		}
+		if int(size) < ones {
```

Comment:
Just realised we also don't check for negative values; `192.168.1.1/-12345` slips through the validation  https://play.golang.org/p/jJM28HwwQcu